helix-core/src/test/java/org/apache/helix/controller/stages/TestIntermediateStateCalcStage.java

```diff
@@ -77,6 +80,11 @@ public void testNoStateMissing() {
           if (resource.endsWith("0")) {
             // Regular recovery balance
             currentStateOutput.setCurrentState(resource, partition, instanceName, "OFFLINE");
+            // add blocked state transition messages
```

Comment:
Could you add some comments about the expected results after adding these pending ST messages? Like partition 'p0' shouldn't be in the output

Reply:
I think you do not understand the expect result. If they are throttled, their final mapping should be their current states.